src/system/SystemPacketBuffer.cpp

```diff
@@ -593,7 +600,11 @@ PacketBufferHandle PacketBufferHandle::New(size_t aAvailableSize, uint16_t aRese
 PacketBufferHandle PacketBufferHandle::NewWithData(const void * aData, size_t aDataSize, size_t aAdditionalSize,
                                                    uint16_t aReservedSize)
 {
+#if INET_CONFIG_ENABLE_TCP_ENDPOINT
+    if (aDataSize > UINT32_MAX)
```

Comment:
Kept this check to ensure data fits within 4 bytes which is the size of the frame length field for messages over TCP.

Reply:
Sure, but you already check in `New()` that the size is not greater than `CHIP_CONFIG_MAX_LARGE_PAYLOAD_SIZE_BYTES`, which is ensured to be within `uint32_t` range. Those checks were needed before only because of the cast to `uint16_t` before calling `New()`.